plugins/renepay/pay.c

```diff
@@ -558,6 +559,7 @@ static struct command_result *try_paying(struct command *cmd,
 						amount_msat_eq(p->total_delivering, AMOUNT_MSAT(0)),
 
 						&err_msg);
+	gossmap_remove_localmods(pay_plugin->gossmap, renepay->local_gossmods);
```

Comment:
I know this is doing similarly to the previous code version. However, there is a potential bug that we should check: hidden channels in the localmods become part of the uncertainty network when `get_payflows` is called we should:
1. make sure they are removed from the uncertainty network once they are not being used anymore, we don't want to bloat the `chan_extra_map` with channels that don't exist anymore.
2. make sure there are no calls `gossmap` functions when we update the knowledge of those channels.

Reply:
Yes: I think we should clean up the ce when we terminate the payment (we want to keep the extra information until then).

I'll do another PR: I want to ship this in -rc2...